server/src/lib/clip.ts

```diff
@@ -252,77 +258,85 @@ export default class Clip {
           console.log("Not converting mpeg/AAC");
        }
 
-      console.log("HEADERS: ", JSON.stringify(headers));
-      const audioOutput = new Transcoder(audioInput)
-        .audioCodec(config.TRANSCODE.AUDIO_CODEC)
-        .format(config.TRANSCODE.FORMAT)
-        .channels(1)
-        //.sampleRate(config.TRANSCODE.SAMPLE_RATE)
-        .on('error', (error: string) => {
-          this.clipSaveError(
-            headers,
-            response,
-            500,
-            `${error} for ${metadata}`,
-            `ffmpeg ${error}`,
-            'clip'
-          );
-          return;
-        })
-        .on('finish', async () => {
-          console.log(`clip written to s3 ${metadata}`);
-
-          await this.model.saveClip({
-            client_id: client_id,
-            localeId: sentence.locale_id,
-            original_sentence_id: sentenceId,
-            path: clipFileName,
-            sentence: sentence.text,
-          });
-          await Awards.checkProgress(client_id, { id: sentence.locale_id });
+      var self = this;
+      promises.mkdtemp(tmpdir() + `/${client_id}_${filePrefix}`)
+          .then(function(tmp_dir_name: string) {
+             const path = tmp_dir_name + `/output.${config.TRANSCODE.FORMAT}`;
+             console.log(path);
+
+             console.log("HEADERS: ", JSON.stringify(headers));
+             console.log(config.TRANSCODE);
+             const audioOutput = new Transcoder(audioInput)
+                .audioCodec(config.TRANSCODE.AUDIO_CODEC)
+                .format(config.TRANSCODE.FORMAT)
+                .channels(1)
+                //.sampleRate(config.TRANSCODE.SAMPLE_RATE)
+                .on('error', (error: string) => {
+                   self.clipSaveError(
+                      headers,
+                      response,
+                      500,
+                      `${error} for ${metadata}`,
+                      `ffmpeg ${error}`,
+                      'clip'
+                   );
+                   return;
+                })
+                .on('finish', async () => {
+                   await self.s3
+                      .upload({
+                         Bucket: config.CLIP_BUCKET_NAME,
+                         Key: clipFileName,
+                         Body: await promises.readFile(path),
+                      })
+                      .promise();
```

Comment:
This is an important one, so maybe we could catch the error here? Similar to the `self.clipSaveError` from above

Reply:
Note that the original code doesn't catch any error.  It's almost a cut and paste.